Task 2/Task 2.1.1/MyTools/MyTools/MyTools.cs

```diff
@@ -0,0 +1,225 @@
+﻿using System;
+
+namespace MyTools
+{
+    public sealed class CustomString : ICloneable
+    {
+        //Fileds
+        private char[] _storage;
+
+        //Constructors
+        private CustomString(int length)
+        {
+            _storage = new char[length];
+        }
+        public CustomString()
+        {
+            _storage = new char[0];
+        }
+        public CustomString(string value)
+        {
+            _storage = new char[value.Length];
+            for (int i = 0; i < _storage.Length; i++) _storage[i] = value[i];
+        }
+        public CustomString(char value, int length)
+        {
+            _storage = new char[length];
+            for (int i = 0; i < _storage.Length; i++) _storage[i] = value;
+        }
+
+        //Properties
+        public static CustomString Empty { get => new CustomString(); }
+        public int Length { get => _storage.Length; }
+
+        //Indexers
+        public char this[int index]
+        {
+            get => (index < 0 || index >= _storage.Length) ? throw new IndexOutOfRangeException(nameof(index)) : _storage[index];
+            set => _storage[(index < 0 || index >= _storage.Length) ? throw new IndexOutOfRangeException(nameof(index)) : index] = value;
```

Comment:
Большие строки получаются. Если используешь тернарки, то лучше тогда переносить каждое вырожение на отдельную строку что то подобие такого:

```suggestion
            get => (index < 0 || index >= _storage.Length)
                               ? throw new IndexOutOfRangeException(nameof(index))
                               : _storage[index];
            set => _storage[(index < 0 || index >= _storage.Length)
                                             ? throw new IndexOutOfRangeException(nameof(index))
                                             : index] = value;
```

Правда тут было излишне кидать исключения, они и так выбросятся если выйти за диапозон значений